app/lib/util/API.dart

```diff
@@ -66,18 +67,22 @@ class SharezoneGateway {
     @required this.schoolClassGateway,
     @required this.schoolGateway,
     @required this.timetable,
+    @required this.user,
   })  : _authUser = authUser,
         uID = authUser.uid,
         userId = UserId(authUser.uid),
         homework = HomeworkGateway(
             userId: authUser.uid,
             firestore: references.firestore,
-            typeOfUserStream: UserGateway(references, authUser)
-                .userStream
-                .map((user) => user?.typeOfUser)),
+            typeOfUserStream: user.userStream.map((user) => user?.typeOfUser)),
         blackboard = BlackboardGateway(
             authUser: authUser, firestore: references.firestore),
         fileSharing =
-            FileSharingGateway(user: authUser, references: references),
-        user = UserGateway(references, authUser);
+            FileSharingGateway(user: authUser, references: references);
+
+  Future<void> dispose() async {
+    await connectionsGateway.dispose();
+    await user.dispose();
+    blackboard.dispose();
```

Comment:
Why is there no await on the last call?

Reply:
I would make the dispose method also async there so it matches the other calls

Reply:
The `dispose` method of `BlackboardGateway` was sync but I saw that we can make it async. Therefore, I changed it to async so that we can await the call 👍